server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
+  const limit = 20;
+  const offset = (page - 1) * limit;
 
-  res.json(users);
+  if (!Number.isInteger(page) || page < 1) {
+    return res.status(400).json({ error: 'Invalid page number' });
+  }
+
+  // Fine all users, using the limit and offset to implement offset pagination
+  const users = await User.findAll({
+    attributes: User.allowedFields,
+    limit: limit,
+    offset: offset,
+  });
+
+  const count = await User.count();
+  const totalPages = Math.ceil(count / limit);
+
+  if (offset >= count && page !== 1) {
```

Comment:
I wanted to exclude 1 so that homepage would not report Page out of range if the DB is empty